core/store/LevelDBStore/LevelDBStore.go

```diff
@@ -509,6 +482,54 @@ func (bd *LevelDBStore) persist(b *Block) error {
 				return err
 			}
 		}
+
+		// init unspent in tx
+		for index := 0; index< len(b.Transactions[i].Outputs); index++ {
+			txhash := b.Transactions[i].Hash()
```

Comment:
ditto, either index<len(b.Transactions[i].Outputs) or index < len(b.Transactions[i].Outputs), but not index< len(b.Transactions[i].Outputs)
we prefer later.